Microsoft.Toolkit.Uwp.UI/ImageCache/ImageCache.cs

```diff
@@ -188,16 +245,18 @@ public static string GetCacheFileName(Uri uri)
             return image;
         }
 
-        private static async Task<BitmapImage> GetFromCacheOrDownloadAsync(Uri uri)
+        private static async Task<BitmapImage> GetFromCacheOrDownloadAsync(Uri uri, string key, bool isPreCache)
         {
             BitmapImage image = null;
             DateTime expirationDate = DateTime.Now.Subtract(CacheDuration);
 
-            var key = GetCacheFileName(uri);
-
             if (MaxMemoryCacheSize > 0)
             {
                 image = GetFromMemoryCache(key);
+                if (isPreCache && image != null)
+                {
+                    return null;
```

Comment:
It doesn't matter, because if isPreCache is true we don't care about the image, we just want it to be in the cache. On the other hand the whole if statement could be omitted I guess, since it doesn't really matter if we return image or null. Let me know how would you like to have it.